chain/index/interface.go

```diff
@@ -0,0 +1,31 @@
+package index
+
+import (
+	"context"
+
+	"github.com/filecoin-project/go-state-types/abi"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/ipfs/go-cid"
+)
+
+// MsgInfo is the Message metadata the index tracks.
+type MsgInfo struct {
+	// the message this record refers to
+	Message cid.Cid
+	// the epoch whre this message was executed
+	Epoch abi.ChainEpoch
+	// the tipset where this messages executed
+	Tipset types.TipSetKey
```

Comment:
Let's use a TipsetCID instead. The TipSetKey is wasteful. It will imply an extra lookup to go from TipsetCid to TipsetKey, but it's worth it to save an average of 4x times the amount of bytes that would otherwise go here, on every record. Discarding a second table + foreign key + joins because that's too much complexity and antithetical to the main goal of speeding things up.

Reply:
done.